src/mqttcm_connect.c

```diff
@@ -427,28 +426,35 @@ void on_connect(struct mosquitto *mosq, void *obj, int reason_code, int flag, co
 	MqttCMInfo("on_connect: success. broker_connect set to 1\n");
 	broker_connect = 1;
 
+	//For Mqtt reconnection state use in-cache subscribe list
 	if(reconnectFlag)
 	{
-		while(1)
+		//printList();
+		comp_topic_name_t* temp = g_head;
+
+		while (temp != NULL)
 		{
-			int subscribeSuccessFlag = 1;
+			MqttCMDebug("Inside mqtt subscribe of %s\n", temp->compName);
+			mqtt_subscribe(temp->compName, temp->topic);
+			temp = temp->next;
+		}
+
+		reconnectFlag = 0;
+	}
+	else  //For crash or process restart case to get the subscribe details from file
+	{
+		if(GetTopicFromFile())
+		{
+			MqttCMInfo("updatetopic from file is successful\n");
```

Comment:
This log is bit confusing as it uses Get func and then update the list, modify log. Also rename the func to GetTopicFromFileandUpdateList() .

Reply:
Insert() also can be renamed to updateToSubscribeList .

Reply:
Insert() do we need?

Reply:
Based on the analysis, it seems we can remove insert function in optimization phase to use only AddToSubscriptionList

Reply:
yes, ok .